src/panels.tsx

```diff
@@ -90,7 +90,7 @@ class PSRouter {
 }
 PS.router = new PSRouter();
 
-class PSRoomPanel<T extends PSRoom = PSRoom> extends preact.Component<{room: T}> {
+class PSRoomPanel<T extends PSRoom = PSRoom, S = {}> extends preact.Component<{room: T}, S> {
```

Comment:
Room panel state should ideally go into the room model (the `PSRoom` instance). This is necessary because Preact doesn't support retaining state when reparenting a component, which we plan to do (partial implementation in #1641).